### PR TITLE
added install macos shell script

### PR DESCRIPTION
**Pull request by micziz**
# Related Issue/Addition to code
No addition to code, and does not fix any issues

- Fixes #
- Issue/Addition to code, goes here.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Proposed Changes
Add a script to automatically install everything on macos

# Additional Info
The script installs home brew, python, git (If not already installed) and then installs required packages.

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by micziz** [comment]
How can i modify the wiki?

File: installmacos.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+fi
+#If no git is found
+if test ! $(which git); then
+    #Install git
+    echo "Installing git..."
+    brew install git
+fi
+
+#Clone the repo
+git clone https://github.com/ThioJoe/YT-Spammer-Purge.git
```
Comment (Firecul) [inline installmacos.sh:24]: I think you should only go until the latest tag (normal or prerelease) as there could be dangling commits that might not be ready for use that could break something potentially.

Comment (micziz) [reply installmacos.sh:24]: Ok, working on that.

Comment (micziz) [reply installmacos.sh:24]: i'm stumped, tring to find a solution, but cant. Any help?

**Comment by micziz** [comment]
This will require extensive documentation rewrite, but i do not know how to edit the wiki. @ThioJoe  will have to do that

**Comment by micziz** [comment]
it installs all required dependencies. It's then up to the user to run it

**Comment by micziz** [comment]
thats why it wloud require extensive wiki rerwite.

**Comment by micziz** [comment]
Or if you want, i can create another repo, with the complete file (using git) and then add that to the wiki

**Comment by micziz** [comment]
> it installs all required dependencies. It's then up to the user to run it

including requirements.txt

**Comment by Firecul** [comment]
> it installs all required dependencies. It's then up to the user to run it

So the user will download the scripts and your install program will install  all the dependencies for them. I think I wasn't understanding originally then.

**Comment by micziz** [comment]
> 

yes!

**Comment by micziz** [comment]
> > it installs all required dependencies. It's then up to the user to run it
> 
> So the user will download the scripts and your install program will install all the dependencies for them. I think I wasn't understanding originally then.

yes!

**Comment by micziz** [comment]
Untill this commit isin't merged, i will create an external repo for downloading the file.

**Comment by EthanHindmarsh** [comment]
Honestly, I'd just add a section to README.md or make an INSTALLING.md -- Normal people can run four commands just fine 😁

**Comment by EthanHindmarsh** [comment]
> thats why it wloud require extensive wiki rerwite.

Not sure what you mean by `extensive`, it'd only be a couple lines?

**Comment by rachmadaniHaryono** [comment]
there is https://github.com/ThioJoe/YT-Spammer-Purge/wiki/MacOS-Instructions

===


non mac user here

any reason why python3.10? 

linux install actually use default python from their package manager

**Comment by EthanHindmarsh** [comment]
As noted above, [Install Instructions for MacOS](https://github.com/ThioJoe/YT-Spammer-Purge/wiki/MacOS-Instructions) are linked on README, I don't see much of a benefit for this

Also, the purpose of the other two install.sh's is to install dependencies from package repositories (from install.sh:)
```
echo "Looks like you aren't running this installer on a Debian, Ubuntu, Fedora, CentOS, or Arch Linux system."
```

From what I can tell, this script makes three assumptions
1. The user downloaded the installer alone -- Odd to say the least?
2. The user wants homebrew installed -- From what I've heard, most probably would, but it's best to not make assumptions
3. The user knows how to run the installer & the python file after -- A user with enough comfort to use the command line wouldn't need the installer. *It may be that you can just double click sh and py files to run, I don't use Mac* 😅

I don't see what problem this really solves. 
If you *can* double click to run sh and py files on MacOS, better to just make a `install_requirements_via_python.sh` (maybe not the greatest name) that just runs `python3 -m ensurepip && python3 -m pip install -r requirements.txt`

**Comment by EthanHindmarsh** [comment]
Might be better to make this a MacOS update script.
As long as it confirms the user wants to install homebrew & git, it'd be fine to pull the latest release

**Comment by rachmadaniHaryono** [comment]
> I don't see much of a benefit for this

i actually can see this working, other project example 

https://spacevim.org/quick-start-guide/

installation instruction

> curl -sLf https://spacevim.org/install.sh | bash

if working both linux and mac osx should have single line install command, for example

> curl -sLf https://raw.githubusercontent.com/micziz/YT-Spammer-Purge/main/install.sh | bash

but this program is different with program above and you still have to update the program either download new package or clone to local folder update to upstream

---

any reason why not add to current install.sh? 

both script will use bash

**Comment by EthanHindmarsh** [comment]
> if working both linux and mac osx should have single line install command, for example

Oh great idea 👍
Didn't consider that at all 😅 
It doesn't seem like the script right now has that purpose though, but if it can be adjusted it'd be great

> Might be better to make this a MacOS update script

Adding an update script along with install would probably still be a useful addition (ex. `git remote get-url origin > /dev/null && git pull` : Check if a remote origin exists and pull )
eg. `install.sh` & `update.sh`
Or it could be combined into one file, just add a check to see if a valid remote exists 
just run the `install.sh` inside the source dir

Combined example:
```bash
# Can be made OS dependent, eg install_macos()
install () {
  echo "installing yt spammer purge"
  # put install code here
}
# Can be made OS dependent, eg update_macos()
update () {
  # maybe check if update needed
  echo "updating your yt spammer purge"
  # update code here
}

# Add checks for OS if we just want one install.sh

git remote get-url origin > /dev/null && update || install
# If get-url succeeds, update, else install
```

**Comment by rachmadaniHaryono** [comment]
>  Or it could be combined into one file, just add a check to see if a valid remote exists

i would vote for this (if install script is used)

something like 

> curl -sLf [micziz/YT-Spammer-Purge@main/install.sh (raw)](https://raw.githubusercontent.com/micziz/YT-Spammer-Purge/main/install.sh) | bash -s -- --update

**Comment by micziz** [comment]
> Adding an update script along with install would probably still be a useful addition (ex. `git remote get-url origin > /dev/null && git pull` : Check if a remote origin exists and pull ) eg. `install.sh` & `update.sh` Or it could be combined into one file, just add a check to see if a valid remote exists just run the `install.sh` inside the source dir
> 
> Combined example:
> 
> ```shell
> # Can be made OS dependent, eg install_macos()
> install () {
>   echo "installing yt spammer purge"
>   # put install code here
> }
> # Can be made OS dependent, eg update_macos()
> update () {
>   # maybe check if update needed
>   echo "updating your yt spammer purge"
>   # update code here
> }
> 
> # Add checks for OS if we just want one install.sh
> 
> git remote get-url origin > /dev/null && update || install
> # If get-url succeeds, update, else install
> ```

Great idea! Litterally did not thing about it

**Comment by EthanHindmarsh** [comment]
> Great idea! Litterally did not thing about it

I'm working on a PR to add that right now 🤷

**Comment by micziz** [comment]
> > Great idea! Litterally did not thing about it
> 
> I'm working on a PR to add that right now 🤷

Ok. gonna close this

**Pull request closed without merging**